code/components/extra-natives-five/src/PatchNetworkChecks.cpp

```diff
@@ -60,13 +63,50 @@ static HookFunction hookFunction([]()
 		hook::put<int32_t>(location, (intptr_t)g_dynamicDoorCreationDisabled - (intptr_t)location - 4);
 	}
 
+	// Replace netgame check in CScenarioConditionIsMultiplayerGame::"check"
+	// This patch was designed to make SP car jacking animations work in MP (CConditionalClipSet)
+	//
+	{
+		uint64_t* result = 0;
+		auto parCguBuildStructureFromStaticData = hook::pattern("48 8D 15 ? ? ? ? 45 33 C9 44 8B C0 48 8B CB E8");
```

Comment:
Patterns without a defined count or count_hint might make initial scan a little slow. Does the result count differ on different game builds?

Reply:
Right, I see what you're doing here now- you're iterating through structure registration routines in order to find `CScenarioConditionIsMultiplayerGame`, and go from _there_ to patch stuff beyond.

I don't have IDA or the game at hand here, but is there any more direct route to get there that won't need _this_?

Alternately, how many matches in does this target usually end up? One could use `.count()` to limit until a bit _after_ that point in time, as order usually matches across executable versions.

Reply:
You could check if `49 2E 5B FA + 0x23c` works (or `04 00 01 00 49 2E 5B FA + 0x240`). Little bit of a hack, but I don't think (verification required) this has changed since 323.

Reply:
hmm something like this would work for 2372/2699, but would need different offset on 1604 I think. It is a pretty hard exact spot to sig.
```
		char* clonedScenarioInfoCtor = hook::get_pattern<char>("48 8D 05 ? ? ? ? 49 89 06 49 8B C6 48 83 C4 30 41 5E C3");
		char* vtable = hook::get_address<char*>(clonedScenarioInfoCtor, 3, 7);
		vtable -= 0xB90;
```

my only nitpick is that several of your asserts dont do anything - it's more likely that if something goes wrong, it's going to just be random junk